### DISCUSSION TITLE
[V5] I need a docs/examples on run custom commands

### DISCUSSION THREAD

**Discussion by alex-oliveira** [adonisjs/core#1941] [Help] [answered]
I created a command using:

`node ace make:command MyCommand`

A command class named 'my:command' has been generated.

**But now how do I do it?**

`node ace my:command`

return:

[ error ]  "my:command" command not found

**Comment by prescience-data** [comment, marked as answer]
Have you added it to your `ace-manifest.json` file?
```json
{
"my:command": {
    "settings": {
      "loadApp": true
    },
    "commandPath": "./commands/MyCommand",
    "commandName": "my:command",
    "description": "A description of your command..."
  },
}
```

**Reply by alex-oliveira** [reply]
It worked! Thank you!

It was not very clear to me that I should manipulate this file :(

**Reply by prescience-data** [reply]
Yes, it seems like the docs for v5 need lots of work but I can imagine it's very hard without a dedicated writer documenting everything. 
Getting the remaining modules ported to TypeScript seems to be the current priority.

**Reply by thetutlage** [reply]
Yeah docs needs love. There is no need to manually add your command. Just run `node ace generate:manifest` and it will update the manifest file by scanning all the commands you project is using

**Reply by AddoSolutions** [reply]
Would it make sense to auto-execute this command after `make:command`?

**Reply by thetutlage** [reply]
Maybe not. Since 90% of the times you will end up changing the command name right after creating the command file and once the command name is changed, you will have to re-generate the manifest file.

However, if you have the watcher running `(node ace serve --watch)`, it will generate the manifest file automatically anytime a new command is added or updated

**Reply by rbartholomay** [reply]
Whats wrong when to command will not be listed after node ace generate:manifest?

I do this steps:

```node ace make:command sample``` 

this will created the folder commands inside the root of my application and in this folder the file Sample.ts

then i do ```node ace generate:manifest```  ...

When i now type ```node ace``` my command is not listed!?

**Reply by rbartholomay** [reply]
... found it...      

   After the upgrade it is also useful to add "./commands" inside the commands-key of the .adonisrc.json  😊

**Reply by bingalls** [reply]
> Yeah docs needs love. There is no need to manually add your command. Just run `node ace generate:manifest` and it will update the manifest file by scanning all the commands you project is using

I tried running this command. After hacking around that `import Application from '@ioc:Adonis/Core/Application'` does not exist, I get an error of `RangeError: Maximum call stack size exceeded`

**Reply by bingalls** [reply]
@rbartholomay Upgrading which to what version solved your issue? I run @adonisjs/ace v6.9.4 and it inserts "./commands" into .adonisrc.json for me. Note that I still cannot run the default command created by `node ace make:command sample`

**Reply by rbartholomay** [reply]
i forgot the step 6 in the upgrade guide: https://preview.adonisjs.com/blog/october-2020-release...    After add this and the key inside adonisrc.json all runs fine in my case

**Comment by bingalls** [comment]
ace-manifest.json does not exist, but https://preview.adonisjs.com/guides/directory-structure#structure-preview tells you to (create) it in the root folder.
I copy the run() method as handle() to suppress a warning. However `node ace my:command` fails with
`TypeError: commandInstance.exec is not a function` and warns of
`Env.getOrFail() is deprecated. Use Env validations instead`
Note that I use node v15.3.0 ; same results with v14.15.0

---
Aha! Fixed by upgrade at https://preview.adonisjs.com/blog/october-2020-release
Still not sure how to upgrade with `@alpha` tags. Change `Env.getOrFail()` in config/app.ts to `Env.get()`

**Reply by thetutlage** [reply]
Watch this video https://www.youtube.com/watch?v=6MBNw6JOaYQ
